src/v/kafka/client/client.cc

```diff
@@ -194,6 +195,14 @@ ss::future<> client::mitigate_error(std::exception_ptr ex) {
         }
     } catch (const ss::gate_closed_exception&) {
         vlog(kclog.debug, "gate_closed_exception");
+    } catch (const std::system_error& ex) {
+        if (ex.code() == std::errc::broken_pipe) {
```

Comment:
this == also checks the category, right?

Reply:
Correct. I originally had it as `switch(static_cast<std::errc>(e.code()))`, which doesn't do that.

Reply:
even seastar bubbles up some non-errno codes (e.g. GnuTLS errors) via system_error (through a different category)